File: patch.py
import sys

def main():
    with open('fresh_supabase_setup.sql', 'r') as f:
        content = f.read()

    with open('supabase/migrations/026_tasks_foundation.sql', 'r') as f:
        tasks_content = f.read()
    
    start_str = "-- 1. Modify point_events table safely"
    end_str = "NOTIFY pgrst, 'reload schema';"
    
    start_idx = tasks_content.find(start_str)
    end_idx = tasks_content.find(end_str)
    if start_idx == -1 or end_idx == -1:
        print("Could not find start/end in tasks migration")
        return
        
    tasks_block = tasks_content[start_idx:end_idx + len(end_str)]
    
    fresh_start_idx = content.find(start_str)
    fresh_end_idx = content.find(end_str, fresh_start_idx)
    
    if fresh_start_idx != -1 and fresh_end_idx != -1:
        new_content = content[:fresh_start_idx] + tasks_block + content[fresh_end_idx + len(end_str):]
        with open('fresh_supabase_setup.sql', 'w') as f:
            f.write(new_content)
        print("Successfully patched fresh_supabase_setup.sql")
    else:
        print("Could not patch fresh_supabase_setup.sql")

if __name__ == '__main__':
    main()

File: patch.ts
import fs from 'fs';

function main() {
    const content = fs.readFileSync('fresh_supabase_setup.sql', 'utf-8');
    const tasksContent = fs.readFileSync('supabase/migrations/026_tasks_foundation.sql', 'utf-8');
    
    const startStr = "-- 1. Modify point_events table safely";
    const endStr = "NOTIFY pgrst, 'reload schema';";
    
    const startIdx = tasksContent.indexOf(startStr);
    const endIdx = tasksContent.indexOf(endStr);
    
    if (startIdx === -1 || endIdx === -1) {
        console.log("Could not find start/end in tasks migration");
        return;
    }
        
    const tasksBlock = tasksContent.substring(startIdx, endIdx + endStr.length);
    
    const freshStartIdx = content.indexOf(startStr);
    const freshEndIdx = content.indexOf(endStr, freshStartIdx);
    
    if (freshStartIdx !== -1 && freshEndIdx !== -1) {
        const newContent = content.substring(0, freshStartIdx) + tasksBlock + content.substring(freshEndIdx + endStr.length);
        fs.writeFileSync('fresh_supabase_setup.sql', newContent);
        console.log("Successfully patched fresh_supabase_setup.sql");
    } else {
        console.log("Could not patch fresh_supabase_setup.sql");
    }
}

main();
